source/extensions/transport_sockets/tls/context_impl.cc

```diff
@@ -714,9 +714,9 @@ bool ContextImpl::dnsNameMatch(const std::string& dns_name, const std::string_vi
       const size_t off = dns_name.length() - pattern_len + 1;
       if (Runtime::runtimeFeatureEnabled("envoy.reloadable_features.fix_wildcard_matching")) {
         return dns_name.substr(0, off).find('.') == std::string::npos &&
-               dns_name.compare(off, pattern_len - 1, pattern.data() + 1, pattern_len - 1) == 0;
+               dns_name.substr(off, pattern_len - 1) == pattern.substr(1, pattern_len - 1);
```

Comment:
Please change the type of `dns_name` to string_view to avoid an allocation here.

Reply:
done